your code is giving vercel build failure 
src/components/Layout.tsx(5,1): error TS6133: 'toggleTheme' is declared but its value is never read.
Error: Command "npm run build" exited with 2
 Have you check in local system before final last pr!

--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "frontend",
   "private": true,
-  "version": "1.1.0",
+  "version": "1.2.0",
   "type": "module",
   "scripts": {
     "setup": "sh scripts/dev-setup.sh",

--- a/src/components/Layout.tsx
+++ b/src/components/Layout.tsx
@@ -2,7 +2,6 @@ import { Outlet } from 'react-router-dom';
 import Navbar from './Navbar';
 import BottomNav from './BottomNav';
 import Footer from './Footer';
-import { toggleTheme } from '../lib/theme';
 
 export default function Layout() {
   return (

--- a/src/components/Navbar.tsx
+++ b/src/components/Navbar.tsx
@@ -91,12 +91,12 @@ export default function Navbar() {
         </div>
 
         {/* Auth Button & Theme Toggle */}
-        <div className="flex items-center gap-4">
+        <div className="flex items-center gap-2 sm:gap-4">
           {/* Theme Toggle Button */}
           <button
             type="button"
             onClick={toggleTheme}
-            className="font-[family-name:var(--font-mono)] text-[10px] tracking-widest text-on-surface-variant hover:text-neon transition-colors duration-200 border border-outline-variant/30 px-3 py-1"
+            className="font-[family-name:var(--font-mono)] text-[9px] sm:text-[10px] tracking-widest text-on-surface-variant hover:text-neon transition-colors duration-200 border border-outline-variant/30 px-2 py-1 sm:px-3"
           >
             THEME
           </button>
@@ -104,7 +104,7 @@ export default function Navbar() {
             <div className="relative" ref={dropdownRef}>
               <button
                 onClick={() => setIsDropdownOpen(!isDropdownOpen)}
-                className="flex items-center gap-3 bg-surface-low border border-outline-variant/30 text-on-surface px-3 py-1.5 transition-all duration-200 hover:bg-surface-mid ghost-border"
+                className="flex items-center gap-2 sm:gap-3 bg-surface-low border border-outline-variant/30 text-on-surface px-2 py-1 sm:px-3 sm:py-1.5 transition-all duration-200 hover:bg-surface-mid ghost-border"
               >
                 {profile?.avatar_url ? (
                   <img src={profile.avatar_url} referrerPolicy="no-referrer" alt="Profile" className="w-7 h-7 object-cover grayscale-[0.5] contrast-125 border border-neon/30" />
@@ -113,8 +113,8 @@ export default function Navbar() {
                     {profile?.full_name?.charAt(0) || 'U'}
                   </div>
                 )}
-                <span className="text-sm font-[family-name:var(--font-mono)] tracking-wider mr-1 uppercase">
-                  {profile?.full_name ? profile.full_name.split(' ')[0] : 'SESSION'}
+                <span className="text-xs sm:text-sm font-[family-name:var(--font-mono)] tracking-wider mr-1 uppercase truncate max-w-[60px] sm:max-w-[100px] inline-block align-bottom" title={profile?.full_name?.split(' ')[0] || 'DEV'}>
+                  {profile?.full_name ? profile.full_name.split(' ')[0] : 'DEV'}
                 </span>
               </button>
 

--- a/src/index.css
+++ b/src/index.css
@@ -105,7 +105,12 @@ body {
 }
 
 /* ── TYPOGRAPHY SYSTEM ── */
-h1, h2, h3, h4, h5, h6 {
+h1,
+h2,
+h3,
+h4,
+h5,
+h6 {
   font-family: var(--font-display);
   font-weight: 700;
   letter-spacing: -0.03em;
@@ -117,9 +122,11 @@ h1, h2, h3, h4, h5, h6 {
 ::-webkit-scrollbar {
   width: 4px;
 }
+
 ::-webkit-scrollbar-track {
   background: var(--color-surface-lowest);
 }
+
 ::-webkit-scrollbar-thumb {
   background: var(--color-outline-variant);
 }
@@ -138,11 +145,9 @@ h1, h2, h3, h4, h5, h6 {
 
 /* ── DOT GRID BACKGROUND ── */
 .dot-grid {
-  background-image: radial-gradient(
-    circle,
-    var(--color-outline-variant) 1px,
-    transparent 1px
-  );
+  background-image: radial-gradient(circle,
+      var(--color-outline-variant) 1px,
+      transparent 1px);
   background-size: 24px 24px;
   opacity: 0.06;
 }
@@ -191,34 +196,80 @@ h1, h2, h3, h4, h5, h6 {
 
 /* ── ANIMATIONS ── */
 @keyframes scan-line {
-  0% { transform: translateY(-100%); }
-  100% { transform: translateY(100%); }
+  0% {
+    transform: translateY(-100%);
+  }
+
+  100% {
+    transform: translateY(100%);
+  }
 }
-.scan-line { animation: scan-line 2.5s ease-in-out infinite; }
+
+.scan-line {
+  animation: scan-line 2.5s ease-in-out infinite;
+}
 
 @keyframes pulse-glow {
-  0%, 100% { box-shadow: 0 0 12px rgba(195, 244, 0, 0.1); }
-  50% { box-shadow: 0 0 28px rgba(195, 244, 0, 0.25); }
+
+  0%,
+  100% {
+    box-shadow: 0 0 12px rgba(195, 244, 0, 0.1);
+  }
+
+  50% {
+    box-shadow: 0 0 28px rgba(195, 244, 0, 0.25);
+  }
 }
-.pulse-glow { animation: pulse-glow 2.5s ease-in-out infinite; }
+
+.pulse-glow {
+  animation: pulse-glow 2.5s ease-in-out infinite;
+}
 
 @keyframes concentric-pulse {
-  0% { transform: scale(1); opacity: 0.6; }
-  100% { transform: scale(2.5); opacity: 0; }
+  0% {
+    transform: scale(1);
+    opacity: 0.6;
+  }
+
+  100% {
+    transform: scale(2.5);
+    opacity: 0;
+  }
 }
 
 @keyframes data-stream {
-  0% { opacity: 0.3; }
-  50% { opacity: 1; }
-  100% { opacity: 0.3; }
+  0% {
+    opacity: 0.3;
+  }
+
+  50% {
+    opacity: 1;
+  }
+
+  100% {
+    opacity: 0.3;
+  }
 }
-.data-stream { animation: data-stream 1.5s ease-in-out infinite; }
+
+.data-stream {
+  animation: data-stream 1.5s ease-in-out infinite;
+}
 
 @keyframes fade-in-up {
-  from { opacity: 0; transform: translateY(20px); }
-  to { opacity: 1; transform: translateY(0); }
+  from {
+    opacity: 0;
+    transform: translateY(20px);
+  }
+
+  to {
+    opacity: 1;
+    transform: translateY(0);
+  }
 }
-.animate-in { animation: fade-in-up 0.6s ease-out forwards; }
+
+.animate-in {
+  animation: fade-in-up 0.6s ease-out forwards;
+}
 
 /* ── VIEWFINDER CORNER BRACKETS ── */
 .viewfinder-corner {
@@ -227,14 +278,43 @@ h1, h2, h3, h4, h5, h6 {
   height: 28px;
   border-color: var(--color-secondary);
 }
-.viewfinder-corner.top-left { top: 0; left: 0; border-top: 2px solid; border-left: 2px solid; }
-.viewfinder-corner.top-right { top: 0; right: 0; border-top: 2px solid; border-right: 2px solid; }
-.viewfinder-corner.bottom-left { bottom: 0; left: 0; border-bottom: 2px solid; border-left: 2px solid; }
-.viewfinder-corner.bottom-right { bottom: 0; right: 0; border-bottom: 2px solid; border-right: 2px solid; }
+
+.viewfinder-corner.top-left {
+  top: 0;
+  left: 0;
+  border-top: 2px solid;
+  border-left: 2px solid;
+}
+
+.viewfinder-corner.top-right {
+  top: 0;
+  right: 0;
+  border-top: 2px solid;
+  border-right: 2px solid;
+}
+
+.viewfinder-corner.bottom-left {
+  bottom: 0;
+  left: 0;
+  border-bottom: 2px solid;
+  border-left: 2px solid;
+}
+
+.viewfinder-corner.bottom-right {
+  bottom: 0;
+  right: 0;
+  border-bottom: 2px solid;
+  border-right: 2px solid;
+}
 
 /* ── FRESHNESS BAR ── */
-.freshness-bar-fresh { border-left: 4px solid var(--color-secondary); }
-.freshness-bar-spoiled { border-left: 4px solid var(--color-error); }
+.freshness-bar-fresh {
+  border-left: 4px solid var(--color-secondary);
+}
+
+.freshness-bar-spoiled {
+  border-left: 4px solid var(--color-error);
+}
 
 /* ── INPUT FIELDS ── */
 input[type="text"],
@@ -270,14 +350,21 @@ input::placeholder {
   letter-spacing: 0.05em;
 }
 
-.no-scrollbar::-webkit-scrollbar { display: none; }
-.no-scrollbar { -ms-overflow-style: none; scrollbar-width: none; }
+.no-scrollbar::-webkit-scrollbar {
+  display: none;
+}
+
+.no-scrollbar {
+  -ms-overflow-style: none;
+  scrollbar-width: none;
+}
 
 /* ── LEAFLET BRUTALIST OVERRIDES ── */
 .leaflet-container {
   background: var(--color-bg-void) !important;
   font-family: var(--font-mono) !important;
 }
+
 .brutalist-popup .leaflet-popup-content-wrapper {
   background: var(--color-surface-mid) !important;
   color: var(--color-on-surface) !important;
@@ -285,6 +372,7 @@ input::placeholder {
   box-shadow: 0 0 0 1px var(--color-outline-variant) !important;
   padding: 0 !important;
 }
+
 .brutalist-popup .leaflet-popup-tip {
   background: var(--color-surface-mid) !important;
   box-shadow: none !important;
@@ -292,8 +380,15 @@ input::placeholder {
   border-top: none !important;
   border-left: none !important;
 }
-.brutalist-popup .leaflet-popup-content { margin: 0 !important; }
-.custom-leaflet-icon { background: transparent !important; border: none !important; }
+
+.brutalist-popup .leaflet-popup-content {
+  margin: 0 !important;
+}
+
+.custom-leaflet-icon {
+  background: transparent !important;
+  border: none !important;
+}
 
 :root.light {
   --bg: #ffffff;

--- a/src/index.css
+++ b/src/index.css
@@ -54,9 +54,21 @@
 }
 
 :root.light {
-  --bg: #ffffff;
+  --bg: #f5f5f5;
   --on-surface: #121212;
   --heading-color: #121212;
+  --color-bg-void: #e8e8e8;
+  --color-surface: #f5f5f5;
+  --color-surface-lowest: #ebebeb;
+  --color-surface-low: #e8e8e8;
+  --color-surface-mid: #e0e0e0;
+  --color-surface-high: #d4d4d4;
+  --color-surface-highest: #cacaca;
+  --color-surface-bright: #c2c2c2;
+  --color-on-surface-variant: #3a3d2e;
+  --color-outline: #5c6048;
+  --color-outline-variant: #b0b59a;
+  --color-tertiary: #121212;
 }
 
 /* ── GLOBAL RESET: ZERO RADIUS ── */

--- a/src/pages/MarketMapPage.tsx
+++ b/src/pages/MarketMapPage.tsx
@@ -7,6 +7,13 @@ import StatusTerminal from '../components/StatusTerminal';
 import { api } from '../lib/api';
 import type { Market } from '../lib/types';
 
+const TILE_DARK  = 'https://{s}.basemaps.cartocdn.com/dark_all/{z}/{x}/{y}{r}.png';
+const TILE_LIGHT = 'https://{s}.basemaps.cartocdn.com/light_all/{z}/{x}/{y}{r}.png';
+
+function getActiveTile() {
+  return document.documentElement.classList.contains('light') ? TILE_LIGHT : TILE_DARK;
+}
+
 function getScoreColor(score: number) {
   return score >= 85 ? 'text-secondary' : score >= 70 ? 'text-neon' : 'text-error';
 }
@@ -37,6 +44,14 @@ export default function MarketMapPage() {
   const [selected, setSelected] = useState<Market | null>(null);
   const [loading, setLoading]   = useState(true);
   const [error, setError]       = useState('');
+  const [tileUrl, setTileUrl] = useState(getActiveTile);
+
+  // Watch for theme class changes on <html> without depending on custom events
+  useEffect(() => {
+    const observer = new MutationObserver(() => setTileUrl(getActiveTile()));
+    observer.observe(document.documentElement, { attributes: true, attributeFilter: ['class'] });
+    return () => observer.disconnect();
+  }, []);
 
   const defaultPosition: [number, number] = [22.5726, 88.3639];
 
@@ -81,10 +96,7 @@ export default function MarketMapPage() {
           scrollWheelZoom={true}
           className="w-full h-full z-0"
         >
-          <TileLayer
-            url="https://{s}.basemaps.cartocdn.com/dark_all/{z}/{x}/{y}{r}.png"
-            attribution="&copy; CARTO"
-          />
+          <TileLayer url={tileUrl} attribution="&copy; CARTO" />
           {markers.map(m => (
             <Marker
               key={m.id}

